client/client.go

```diff
@@ -102,8 +109,9 @@ func (c *packetConn) WriteTo(b []byte, addr net.Addr) (int, error) {
 	if socksTargetAddr == nil {
 		return 0, errors.New("Failed to parse target address")
 	}
-	cipherBuf := newUDPBuffer()
-	defer freeUDPBuffer(cipherBuf)
+	box := udpPool.Slice()
```

Comment:
The variable is called `box`, but the method `Slice()`. You may want to agree on one name.

Reply:
Perhaps we call this something like LazySlice, to not confuse with a standard slice?

Reply:
Done